File: Daily Challanges/2026-03-March/2026-03-05.py
def smallest_gap(s):
    smallest_substring = s
    for i in range(len(s) - 1):
        for j in range(i+1, len(s)):
            if s[i] == s[j]:
                substring = s[i+1:j]
                n = len(substring)
                m = len(smallest_substring)
                if n < m:
                    smallest_substring = substring
    return smallest_substring

test1 = smallest_gap("ABCDAC")
test2 = smallest_gap("racecar")
test3 = smallest_gap("A{5e^SD*F4i!o#q6e&rkf(po8|we9+kr-2!3}=4")
test4 = smallest_gap("Hello World")
test5 = smallest_gap("The quick brown fox jumps over the lazy dog.")

check1 = "DA"
check2 = "e"
check3 = "#q6e&rkf(p"
check4 = ""
check5 = "fox"

print(f"Test1: {test1 == check1}")
print(f"Test2: {test2 == check2}")
print(f"Test3: {test3 == check3}")
print(f"Test4: {test4 == check4}")
print(f"Test5: {test5 == check5}")

File: Daily Challanges/2026-03-March/2026-03-05.js
function smallestGap(str) {
  const lastSeen = {};
  let minGap = str.length;
  let smallestSubstring = "";
  for (let i = 0; i < str.length; i++) {
    const char = str[i];
    if (lastSeen.hasOwnProperty(char)) {
      const gap = i - lastSeen[char];
      if (gap < minGap) {
        minGap = gap;
        smallestSubstring = str.slice(lastSeen[char] + 1, i);
      }
    }
    lastSeen[char] = i;
  }
  return smallestSubstring;
}

const test1 = smallestGap("ABCDAC");
const test2 = smallestGap("racecar");
const test3 = smallestGap("A{5e^SD*F4i!o#q6e&rkf(po8|we9+kr-2!3}=4");
const test4 = smallestGap("Hello World");
const test5 = smallestGap("The quick brown fox jumps over the lazy dog.");

const check1 = "DA";
const check2 = "e";
const check3 = "#q6e&rkf(p";
const check4 = "";
const check5 = "fox";

console.log(`Test1: ${test1 === check1}`);
console.log(`Test2: ${test2 === check2}`);
console.log(`Test3: ${test3 === check3}`);
console.log(`Test4: ${test4 === check4}`);
console.log(`Test5: ${test5 === check5}`);
